tony-history-server/test/utils/TestHdfsUtils.java

```diff
@@ -76,38 +77,106 @@ public void testContentOfHdfsFile_throwsException() throws IOException {
-    String histFolder = "./test/resources/typicalHistFolder";
+  public void testGetJobId_emptyPath() {
+    Path filePath = mock(Path.class);
+    when(filePath.toString()).thenReturn("");
+
+    assertEquals("", HdfsUtils.getJobId(filePath.toString()));
+  }
+
+  @Test
+  public void testIsJobFolder_match() {
+    Path filePath1 = new Path("/abc/def/application_1541469337545_0024");
+    Path filePath2 = new Path("/abc/def/job2");
+    String regex1 = "^application_.*";
+    String regex2 = "^job.*";
+
+    assertTrue(HdfsUtils.isJobFolder(filePath1, regex1));
+    assertTrue(HdfsUtils.isJobFolder(filePath2, regex2));
+  }
+
+  @Test
+  public void testIsJobFolder_notMatch() {
+    Path filePath1 = new Path("./test/job/application_1541469337545_0024");
+    Path filePath2 = new Path("./test/resources/application2/");
+    String regex1 = ".*job.*";
+    String regex2 = "application_.*";
+
+    assertFalse(HdfsUtils.isJobFolder(filePath1, regex1));
+    assertFalse(HdfsUtils.isJobFolder(filePath2, regex2));
+  }
+
+  @Test
+  public void testGetJobFolders_emptyHistoryFolder() {
+    Path histFolder = new Path("./test/resources/emptyHistFolder");
+
+    assertEquals(new ArrayList<Path>(), HdfsUtils.getJobFolders(fs, histFolder, "job*"));
+  }
+
+  @Test
+  public void testGetJobFolders_typicalHistoryFolder() {
+    Path histFolder = new Path("./test/resources/typicalHistFolder");
+    String regex = "^job.*";
     List<Path> expectedRes = new ArrayList<>();
 
     for (int i = 1; i < 6; ++i) {
       StringBuilder sb = new StringBuilder();
       sb.append("file:").append(System.getProperty("user.dir"));
       sb.append("/test/resources/typicalHistFolder/job").append(i);
-      sb.append("/metadata.json");
       expectedRes.add(new Path(sb.toString()));
     }
-    List<Path> actualRes = HdfsUtils.getMetadataFilePaths(fs, histFolder);
+    List<Path> actualRes = HdfsUtils.getJobFolders(fs, histFolder, regex);
     Collections.sort(actualRes);
 
     assertEquals(expectedRes, actualRes);
     assertEquals(expectedRes.size(), actualRes.size());
   }
 
   @Test
-  public void testGetMetadataFilePaths_throwsException() throws IOException {
-    String histFolder = "./test/resources/typicalHistFolder";
+  public void testGetJobFolders_nestedHistFolder() {
+    Path histFolder = new Path("./test/resources/nestedHistFolder");
+    String regex = "^job.*";
+    List<Path> expectedRes = new ArrayList<>();
+    StringBuilder sb = new StringBuilder();
+    sb.append("file:").append(System.getProperty("user.dir"));
+    sb.append("/test/resources/nestedHistFolder/");
+    String baseDir = sb.toString();
+
+    expectedRes.add(new Path(baseDir, "2018/01/02/job0"));
+    expectedRes.add(new Path(baseDir,"2018/01/01/job1"));
+    expectedRes.add(new Path(baseDir,"2018/12/31/job2"));
+    expectedRes.add(new Path(baseDir,"2017/07/job3"));
+    expectedRes.add(new Path(baseDir,"2017/07/job4"));
+    expectedRes.add(new Path(baseDir,"2017/07/job5"));
```

Comment:
Nit: please add space after `baseDir,`

(We should enable Checkstyle on our code. Ref: https://docs.gradle.org/current/userguide/checkstyle_plugin.html)

Reply:
👍 We can do this in the following PRs. Right now, I'm just gonna manually fix these.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/81/commits/108fe8650f76b5d9d5ae71188f3b95d170b54a93.